homeassistant/components/jewish_calendar/services.yaml

```diff
@@ -0,0 +1,241 @@
+get_gregorian_date:
+  fields:
+    date:
+      example: 2023-01-01
+      selector:
+        date:
+    include_hebrew_date_info: &include_hebrew_date_info
+      default: true
+      selector:
+        boolean:
+    include_holiday_info: &include_holiday_info
+      selector:
+        boolean:
+    include_zmanim: &include_zmanim
+      selector:
+        boolean:
+
+get_hebrew_date:
+  fields:
+    year:
+      example: 5780
+      selector:
+        number:
+          mode: box
+    month:
+      example: Tishrei
+      selector:
+        select:
+          options:
+            - Tishrei
+            - Marcheshvan
+            - Kislev
+            - Tevet
+            - Sh'vat
+            - Adar I
+            - Adar II
+            - Nisan
+            - Iyyar
+            - Sivan
+            - Tammuz
+            - Av
+            - Elul
+            - תשרי
+            - מרחשוון
+            - כסלו
+            - טבת
+            - שבט
+            - אדר א
+            - אדר ב
+            - ניסן
+            - אייר
+            - סיון
+            - תמוז
+            - אב
+            - אלול
+    day:
+      example: 1
+      selector:
+        number:
+          min: 1
+          max: 30
+          mode: box
+    include_hebrew_date_info: *include_hebrew_date_info
+    include_holiday_info: *include_holiday_info
+    include_zmanim: *include_zmanim
+
+get_next_holiday:
+  fields:
+    date:
+      example: 2023-01-01
+      selector:
+        date:
+    types: &types_field
+      selector:
+        select:
+          multiple: true
+          options:
+            - YOM_TOV
+            - EREV_YOM_TOV
+            - HOL_HAMOED
+            - MELACHA_PERMITTED_HOLIDAY
+            - FAST_DAY
+            - MODERN_HOLIDAY
+            - MINOR_HOLIDAY
+            - MEMORIAL_DAY
+            - ISRAEL_NATIONAL_HOLIDAY
+    include_hebrew_date_info: *include_hebrew_date_info
+    include_holiday_info: *include_holiday_info
+    include_zmanim: *include_zmanim
+get_holidays:
+  fields:
+    year:
+      example: 5780
+      selector:
+        number:
+          mode: box
+    types: *types_field
+    holidays:
+      selector:
+        select:
+          multiple: true
+          options:
+            - erev_rosh_hashana
+            - Erev Rosh Hashana
```

Comment:
Why are these duplicated in pairs like this; the first item looks like a key, the second like a label?

Reply:
For maximal flexibility, I wanted users to be able to specify values from any of the columns in https://www.home-assistant.io/integrations/jewish_calendar/#holiday-sensor (as per https://en.wikipedia.org/wiki/Robustness_principle)

The IDs are more useful for complex scripts / templating, so I think it'd be fine to remove them from this YAML (which AFAIK only affects the Call Service UI; not the actual validation) if you want.

If you only want to list one language here, I would need to move this logic to YAML (as per the other discussion) so that it can consume the language options from the config flow.

Reply:
I think it would be much cleaner to allow specifying only the ID and provide translations for the select selector in Hebrew  or English according to the user's setting. That means you'd dynamically create the service schema instead of hard coding it in services.yaml, correct?

Reply:
Done (and tested)